src/code/z_horse.c

```diff
@@ -1,21 +1,233 @@
+#include "z64horse.h"
 #include "global.h"
+#include "overlays/actors/ovl_Bg_Umajump/z_bg_umajump.h"
+#include "overlays/actors/ovl_En_Horse/z_en_horse.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_horse/func_800F3940.s")
+s32 func_800F3940(PlayState* play) {
+    Actor* actor;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_horse/func_800F39B4.s")
+    for (actor = NULL;; actor = actor->next) {
+        actor = SubS_FindActor(play, actor, ACTORCAT_PROP, ACTOR_BG_UMAJUMP);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_horse/func_800F3A64.s")
+        if (actor == NULL) {
+            break;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_horse/func_800F3B2C.s")
+        if (actor->params == BG_UMAJUMP_TYPE_2) {
+            return ((BgUmajump*)actor)->objectIndex;
```

Comment:
Looking at BgUmajump_Init, objectIndex is probably better as a union (or possibly just renamed to index since it seems to store either a PathIndex or an ObjectIndex). For `BG_UMAJUMP_TYPE_2` the value from the params never gets overwritten (`BG_UMAJUMP_GET_OBJECT_INDEX` seems misnmaed, probably is a `PathIndex`). (side note could probably rename `BG_UMAJUMP_GET_FF` as `BG_UMAJUMP_GET_TYPE`).

Reply:
`objectIndex` is currently an union! BgUmajump uses that member as a timer too
```c
    /* 0x160 */ union {
                    s32 objectIndex;
                    s32 rotationTimer; // y rotation
                };
```

And yeah, I think you are right, `BG_UMAJUMP_GET_OBJECT_INDEX` should probably be renamed to `GET_PATH_INDEX`, I'll do that